crates/telemetry-subscribers/src/lib.rs

```diff
@@ -272,14 +392,12 @@ impl TelemetryConfig {
                 .with_ansi(config.log_file.is_none() && stderr().is_tty())
                 .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
                 .with_writer(nb_output)
+                .with_filter(log_filter)
                 .boxed();
             layers.push(fmt_layer);
         }
 
-        tracing_subscriber::registry()
-            .with(filter)
-            .with(layers)
-            .init();
```

Comment:
I actually think that the separate "span_level" knob could be completely eliminated if we restructured the way that our subscriber layers are built. Today we stick the filter first and then all of the layers in parallel. instead we could do something like:

```
tracing_subscriber::registry()
            .with(tokio_console) // if enabled
            .with(filter)
            .with(layers)
```

That way the filter is applied after tokio console.

Reply:
We could do that, but the point here is that we _want_ a separate knob.  We don't want the same knob for both logs and span-related stuff, that doesn't work at all.  For example we want automatic span latencies and tracing for spans down to trace level (if tracing is turned on), but we definitely don't want to log at TRACE level.